src/ui/public/share/directives/share.js

```diff
@@ -1,18 +1,109 @@
+import {
+  getUnhashableStatesProvider,
+  unhashUrl,
+} from 'ui/state_management/state_hashing';
+
+import urlShortenerProvider from '../lib/url_shortener';
+
 import uiModules from 'ui/modules';
 import shareTemplate from 'ui/share/views/share.html';
 const app = uiModules.get('kibana');
 
-app.directive('share', function () {
+app.directive('share', function (Private, Notifier) {
+  const getUnhashableStates = Private(getUnhashableStatesProvider);
+  const urlShortener = Private(urlShortenerProvider);
+
   return {
     restrict: 'E',
     scope: {
       objectType: '@',
       objectId: '@',
-      setAllowEmbed: '&?allowEmbed'
+      allowEmbed: '@allowEmbed'
     },
     template: shareTemplate,
-    controller: function ($scope) {
-      $scope.allowEmbed = $scope.setAllowEmbed ? $scope.setAllowEmbed() : true;
+    controllerAs: 'share',
+    controller: function ($scope, $location, globalState) {
+      // Default to allowing an embedded IFRAME, unless it's explicitly set to false.
+      this.allowEmbed = $scope.allowEmbed === 'false' ? false : true;
```

Comment:
I'm coming to terms with the fact that `@` is a bit more predictable in this scenario, especially if we aren't going to bind to these values over time. Do you think it would make sense to validate this scope value a bit though?